src/libcrun/kontain_nonkmexec.c

```diff
@@ -0,0 +1,343 @@
+/*
+ * crun - OCI runtime written in C
+ *
+ * Copyright (C) 2020 Kontain Inc.
+ * crun is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * crun is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with crun.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <pthread.h>
+#include <string.h>
+#include <time.h>
+#include <sys/queue.h>
+#include <regex.h>
+#include <stdlib.h>
+#include <errno.h>
+#include "utils.h"
+#include "sha256.h"
+
+/*
+ * What is going on in here?
+ * We need to allow non kontain programs to be run inside of a kontain container.
```

Comment:
it has not really "not kontain program". It is simply "Sometimes (e.g. for troubleshooting) we may need to run an executable in container context, but without Kontain VM encapsulation."

Reply:
I hesitate to speculate on the reasons people do things with programs.
I've found that people use programs in completely different ways than the designers intended if it helps them get their jobs done.
And, they will probably use this feature for their own purposes too.